src/sssd.py

```diff
@@ -103,19 +112,34 @@ def save_ca_cert(ca_cert: str) -> None:
         logger.error(f"{e} Reason:\n{e.stderr}")
 
 
-def save_conf(sssd_conf: str) -> None:
+def save_conf(
+    basedn: str, domain: str, ldap_uri: str, ldap_default_bind_dn: str, ldap_password: str
+) -> None:
     """Save sssd conf.
 
     Args:
-        sssd_conf: SSSD configuration file.
+        basedn:   Default base DN.
+        domain:   Domain name.
+        ldap_uri: LDAPS address.
+        ldap_default_bind_dn: Default bind DN from secret.
+        ldap_password: Password passed from secret.
     """
-    sssd_conf_path = "/etc/sssd/sssd.conf"
-    # Decode base64 string and writes to path
-    _save(sssd_conf, sssd_conf_path)
+    sssd_conf_path = "/etc/sssd/conf.d/sssd.conf"
+    # Write contents to config file
+    with open("templates/sssd.toml.j2", "r") as f:
+        template = Template(f.read())
```

Comment:
What about this instead?

```suggestion
    template = Template(pathlib.Path("templates/sssd.toml.j2").read_text())
```